packages/database/__tests__/services/migrations/migrationHooks.test.ts

```diff
@@ -0,0 +1,216 @@
+    allTables = (
+      await database.sequelize.query(
+        "SELECT table_schema as schema, table_name as table FROM information_schema.tables where table_schema IN ('public', 'logs') and table_type != 'VIEW'",
+      )
+    )[0].filter(({ schema, table }) => !GLOBAL_EXCLUDE_TABLES.includes(`${schema}.${table}`));
+  });
+
+  afterEach(async () => {
+    await removeFakeTriggerFunction(database.sequelize);
+    await closeDatabase();
+  });
+
+  describe('tablesWithTrigger', () => {
+    it('should return no tables for non-existing trigger', async () => {
+      const tables = await tablesWithTrigger(database.sequelize, 'banana_', '_chocolate', []);
+      expect(tables).toEqual([]);
+    });
+
+    it('should return tables that have a trigger', async () => {
+      const prefix = 'cat_';
+      const suffix = '_dog';
+      const triggeredTables = randomSelection(allTables);
+      for (const table of triggeredTables) {
+        await addTrigger(database.sequelize, table, prefix, suffix);
+      }
+
+      const tables = await tablesWithTrigger(database.sequelize, prefix, suffix, []);
+
+      for (const table of triggeredTables) {
+        await removeTrigger(database.sequelize, table, prefix, suffix);
+      }
+
+      expect(tables.sort(sortTables)).toEqual(triggeredTables.sort(sortTables));
+    });
+
+    it('should return tables that have a trigger even if it exceeds trigger character limit', async () => {
+      const prefix = 'super_duper_long_trigger_name_';
+      const suffix = '_really_silly_ridiculous_crazy_long_suffix';
+      const triggeredTables = randomSelection(allTables);
+      for (const table of triggeredTables) {
+        await addTrigger(database.sequelize, table, prefix, suffix);
+      }
+
+      const tables = await tablesWithTrigger(database.sequelize, prefix, suffix, []);
+
+      for (const table of triggeredTables) {
+        await removeTrigger(database.sequelize, table, prefix, suffix);
+      }
+
+      expect(tables.sort(sortTables)).toEqual(triggeredTables.sort(sortTables));
+    });
+
+    it('should not return tables that are flagged in the exclude list', async () => {
+      const prefix = 'bird_';
+      const suffix = '_bat';
+      const triggeredTables = randomSelection(allTables);
+      for (const table of triggeredTables) {
+        await addTrigger(database.sequelize, table, prefix, suffix);
+      }
+
+      const excludeTables = randomSelection(allTables);
+      const tables = await tablesWithTrigger(
+        database.sequelize,
+        prefix,
+        suffix,
+        excludeTables.map(({ schema, table }) => `${schema}.${table}`),
+      );
+
+      for (const table of triggeredTables) {
+        await removeTrigger(database.sequelize, table, prefix, suffix);
+      }
+
+      expect(tables.sort(sortTables)).toEqual(
+        triggeredTables.filter(table => !excludeTables.includes(table)).sort(sortTables),
+      );
+    });
+  });
+
+  describe('tablesWithoutTrigger', () => {
```

Comment:
This test suite is almost identical to the above one. I could do the `jest.each` approach, but tbh I've never loved templating in unit tests so I don't mind being WET here. But happy if reviewer wants me to refactor.